MiniAn/minian_parameters.py

```diff
@@ -0,0 +1,335 @@
+        self.parameters  = danse_parameters.get(mn_defs.ParametersKeys.PARAMETERS, {})
+        self.preview = False
+        if mn_defs.ParametersKeys.PREVIEW in danse_parameters:
+            self.preview = True
+            self.preview_parameters = danse_parameters[mn_defs.ParametersKeys.PREVIEW]
+
+        os.environ["OMP_NUM_THREADS"] = "1"
+        os.environ["MKL_NUM_THREADS"] = "1"
+        os.environ["OPENBLAS_NUM_THREADS"] = "1"
+        os.environ["MINIAN_INTERMEDIATE"] = os.path.join(self.paths[mn_defs.ParametersKeys.TMP_DIR], "intermediate")
+
+        parameters = self.parameters
+
+        self.fr = utils.get_frequency(self.paths[mn_defs.ParametersKeys.FNAME], self.paths[mn_defs.ParametersKeys.H5PATH]+'Time')
+
+        neuron_diameter     = tuple((np.array([self.parameters[mn_defs.ParametersKeys.NEURO_DIAM_MIN], self.parameters[mn_defs.ParametersKeys.NEURO_DIAM_MAX]])/self.parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP]).round().astype('int'))
+        noise_freq: float   = parameters[mn_defs.ParametersKeys.NOISE_FREQ]
+        thres_corr: float   = parameters[mn_defs.ParametersKeys.THRES_CORR]
+
+        advanced_settings = parameters.get(mn_defs.ParametersKeys.ADVANCED_SETTINGS, {})
+
+        # removing advanced_sesttings function keys that are not in the minian functions list
+        minian_functions_list = ["TaskAnnotation", "get_optimal_chk", "custom_arr_optimize",
+                                "save_minian", "open_minian", "denoise", "remove_background",
+                                "seeds_init", "pnr_refine", "ks_refine", "seeds_merge", "initA", "initC",
+                                "compute_trace", "get_noise_fft", "update_spatial", "update_temporal",
+                                "unit_merge", "update_background", "compute_AtC", "apply_transform",
+                                "estimate_motion"] + ["LocalCluster"]
+
+        advanced_settings = {key: advanced_settings[key] for key in advanced_settings if key in minian_functions_list}
+
+        self.params_LocalCluster = {
+            "n_workers": 4,
+            "memory_limit": "auto",
+            "resources": {"MEM": 1}, # constrain the number of tasks that can be concurrently in memory for each worker
+            "threads_per_worker": 2,
+            "dashboard_address": ":8787",
+            "local_directory": self.paths[mn_defs.ParametersKeys.TMP_DIR]
+        }
+        if "LocalCluster" in advanced_settings:
+            advanced_settings["LocalCluster"] = {key: advanced_settings["LocalCluster"][key] for key in advanced_settings["LocalCluster"] if key in self.params_LocalCluster}
+            self.params_LocalCluster.update(advanced_settings["LocalCluster"])
+
+        self.params_load_doric = {
+            "fname": self.paths[mn_defs.ParametersKeys.FNAME],
+            "h5path": self.paths[mn_defs.ParametersKeys.H5PATH],
+            "dtype": np.uint8,
+            "downsample": {"frame": parameters["TemporalDownsample"],
```

Comment:
Isn't "TemporalDownsample" in definitions?

Reply:
forgot this one :(